src/display/dom_utils.js

```diff
@@ -33,6 +33,37 @@ var createValidAbsoluteUrl = sharedUtil.createValidAbsoluteUrl;
 
 var DEFAULT_LINK_REL = 'noopener noreferrer nofollow';
 
+function DOMCanvasFactory() {}
+DOMCanvasFactory.prototype = {
+  create: function DOMCanvasFactory_create(width, height) {
+    var canvas = document.createElement('canvas');
+    canvas.width = width || 0;
+    canvas.height = height || 0;
+
+    return canvas;
+  },
+
+  reset: function DOMCanvasFactory_reset(canvas, width, height) {
+    if (!canvas) {
```

Comment:
Use `assert(canvas, 'canvas is not specifed')` here (and below)

add asserts (here and above) for the width and height parameters: `assert(width > 0 && height > 0, 'invalid canvas size')`